Modix/Modules/BetModule.cs

```diff
@@ -0,0 +1,230 @@
+﻿using Discord;
+using Discord.Commands;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Modix.Modules
+{
+    [Group("bet"), Name("Bet"), Summary("A fun betting game involving random numbers!")]
+    public class BetModule : ModuleBase
+    {
+        private enum BettingState
+        {
+            Idle,
+            Stage,
+            Rolling,
+            Results
+        }
+
+        private struct UserBag
+        {
+            public ulong Id { get; set; }
+            public string UserName { get; set; }
+        }
+
+        private const int StageDurationInSeconds = 30;
+        private const int StageStepLengthInSeconds = 5;
+        private const int RollingDurationInSeconds = 30;
+        private const int RollingStepLengthInSeconds = 5;
+
+        private static int _stageCounter;
+        private static int _rollingCounter;
+        private static IUser _sessionOwner;
+        private static BettingState _state;
+        private static Timer _stageTimer;
+        private static Timer _rollingTimer;
+        private static List<IUser> _joinedUsers = new List<IUser>();
+        private static Dictionary<IUser, int> _rolls = new Dictionary<IUser, int>();
+        private static Random _randomGenerator = new Random();
+
+        [Command("Stage"), Summary("Sets the stage for a new betting game. Players can join during this time with !bet join")]
+        public async Task StageAsync()
+        {
+            if (_state != BettingState.Idle)
```

Comment:
Can we invert this to remove nesting? looks like it, i cant stage something if its idle